interfaces/acados_template/acados_template/c_templates_tera/matlab_templates/acados_mex_set_p_global.in.c

```diff
@@ -0,0 +1,120 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.;
+ */
+
+
+// standard
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// acados
+#include "acados/utils/print.h"
+#include "acados_c/ocp_nlp_interface.h"
+#include "acados_solver_{{ name }}.h"
+{% if problem_class == 'OCP' and dims.np_global > 0 %}
+#include "{{ model.name }}_model/{{ model.name }}_model.h"
+#include "{{ name }}_p_global_precompute_fun.h"
+{% elif problem_class == 'MOCP' and phases_dims[0].np_global > 0 %}
+{%- for jj in range(end=n_phases) %}
+#include "{{ model[jj].name }}_model/{{ model[jj].name }}_model.h"
+{% endfor %}
+#include "{{ name }}_p_global_precompute_fun.h"
+{% endif %}
+// mex
+#include "mex.h"
+#include "mex_macros.h"
+
+
+void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
+{
+
+    long long *ptr;
+    char fun_name[20] = "ocp_set_p_global";
+    char buffer [500]; // for error messages
+
+    /* RHS */
+    int nrhs_ref = 2;
+
+    // C ocp
+    const mxArray *C_ocp = prhs[0];
+    // capsule
+    ptr = (long long *) mxGetData( mxGetField( C_ocp, 0, "capsule" ) );
+    {{ name }}_solver_capsule *capsule = ({{ name }}_solver_capsule *) ptr[0];
+    // dims
+    ptr = (long long *) mxGetData( mxGetField( C_ocp, 0, "dims" ) );
+    ocp_nlp_dims *dims = (ocp_nlp_dims *) ptr[0];
+
+    // value
+    double *value = mxGetPr( prhs[1] );
+
+    // for checks
+    int matlab_size = (int) mxGetNumberOfElements( prhs[1] );
+    int nrow = (int) mxGetM( prhs[1] );
+    int ncol = (int) mxGetN( prhs[1] );
+
+    if (nrhs == nrhs_ref)
+    {
+    {% if problem_class == 'OCP' and dims.np_global > 0 or problem_class == 'MOCP' and phases_dims[0].np_global > 0 %}
+        external_function_casadi* fun = &capsule->p_global_precompute_fun;
+        fun->args[0] = value;
+    {% if problem_class == 'OCP' and dims.np_global > 0 %}
+        int np_global = {{ dims.np_global }};
+    {% elif problem_class == 'MOCP' and phases_dims[0].np_global > 0 %}
+        int np_global = {{ phases_dims[0].np_global }};
+    {%- endif %}
+        if (matlab_size != np_global)
+        {
+            sprintf(buffer, "acados_mex_set_p_global: np_global = %d should match data_len = %d. Exiting.\n", np_global, matlab_size);
+            mexErrMsgTxt(buffer);
+        }
+
+    {% set n_pools = casadi_pool_names | length %}
```

Comment:
I am quite confused about this.
It should not be needed and we should just call the function:
`int {{ name }}_acados_set_p_global({{ name }}_solver_capsule* capsule, double* data, int data_len)`
provided in `acados_solver.in.c` and with the same signature in the multi-phase template.
Or am I missing something?

Finally, we could completely avoid the file, if we implement it within `ocp_set`.
Which I think it is a good idea to try to not add too many templated MEX files.